code/src/components/products-container/ProductsContainer.tsx

```diff
@@ -36,8 +45,18 @@ const ProductsContainer = ({
 
   const skeletonCards = repeatComponent(<ProductSkeleton />, loadingItemsCount);
 
-  const handleAddToCart = () => {
-    openDrawer(<CartDrawer />);
+  const handleAddToCart = async (product: Product) => {
+    try {
+      if (user?.id) {
+        await addToCart({ productId: product.id, userId: user.id }).unwrap();
+        openDrawer(<CartDrawer />);
```

Comment:
Why do we use `unwrap` here?

Reply:
Because of `unwrap`, result is returned immediately. In this case I need error to be thrown for `try catch` block to work. Without `unwrap` error is handled internally and `try catch` block would not work.